action_server/frontend/src/views/runs/index.tsx

```diff
@@ -32,9 +32,20 @@ import { ActionRunsContext, useActionRunsContext } from './components/context';
 import { ActionRunDetails } from './components/ActionRunDetails';
 
 const RunRow: FC<TableRowProps<RunTableEntry>> = ({ rowData: run }) => {
-  const { setShowRun } = useActionRunsContext();
+  const { setShowRun, showRun } = useActionRunsContext();
   const navigate = useNavigate();
 
+  // Issue: if we have a run and click it, we'll set the showRun to the run,
+  // but afterwards, if the run is changed (so, we have a new run instance
+  // representing the same run), we have to use an effect to update the clicked
+  // run.
+
+  useEffect(() => {
+    if (run && showRun && run.id === showRun.id) {
+      setShowRun(run);
+    }
+  }, [run, showRun]);
```

Comment:
would setShowRun ever change? if so, may be worth adding here as well to prevent using stale version of it

Reply:
It shouldn't, but no harm in adding it either, so, I'll change to add that too, who knows if it'll change in the future.